src/raw_ptr.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{
+    db::{DBInner, DB},
+    db_options::Options,
+    env::Env,
+};
+use librocksdb_sys as ffi;
+
+/// Trait for accessing raw pointers to underlying RocksDB objects.
+///
+/// This trait is only available when the `raw-ptr` feature is enabled.
+/// It provides access to the underlying C API pointers for advanced use cases.
+///
+/// # Examples
+///
+/// ```rust
+/// use rocksdb::{DB, Options, AsRawPtr};
+///
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let db = DB::open_default("path")?;
+/// let raw_db_ptr = db.as_raw_ptr(); // *mut rocksdb_t
+///
+/// let opts = Options::default();
+/// let raw_opts_ptr = opts.as_raw_ptr(); // *mut rocksdb_options_t
+///
+/// // You can now use these pointers with the C API directly
+/// // unsafe { rocksdb_some_c_function(raw_db_ptr, raw_opts_ptr); }
+/// # Ok(())
+/// # }
+/// ```
+///
+/// # Safety
+///
+/// The returned pointers are only valid as long as the Rust objects are alive.
+/// You must ensure proper lifetime management and avoid using the pointers
+/// after the objects have been dropped.
+pub trait AsRawPtr<T> {
+    /// Returns a raw pointer to the underlying RocksDB object.
+    ///
+    /// # Safety
+    ///
+    /// The returned pointer is only valid as long as the object implementing
+    /// this trait is alive. The caller must ensure proper lifetime management
+    /// and avoid using the pointer after the object has been dropped.
+    fn as_raw_ptr(&self) -> *mut T;
```

Comment:
nit: maybe best to mark as unsafe to really really make sure the user knows they are in the danger zone